src/request_id.rs

```diff
@@ -25,12 +25,24 @@ use uuid::Uuid;
 ///   format!("{}", uuid)
 /// }
 /// ```
+///
+/// Optionally, using the `uuid_v7` feature flag will allow [`RequestId`] to use UUID v7 instead of the currently used UUID v4.
+///
+/// However, the [`uuid`] crate requires a compile time flag `uuid_unstable` to be passed in `RUSTFLAGS="--cfg uuid_unstable"` in order to compile. You can read more about it [here](https://docs.rs/uuid/latest/uuid/#unstable-features).
+///
 #[derive(Clone, Copy, Debug)]
 pub struct RequestId(Uuid);
 
 impl RequestId {
     pub(crate) fn generate() -> Self {
-        Self(Uuid::new_v4())
+        #[cfg(not(feature = "uuid_v7"))]
+        {
+            Self(Uuid::new_v4())
+        }
+        #[cfg(all(uuid_unstable, feature = "uuid_v7"))]
+        {
+            Self(Uuid::now_v7())
+        }
```

Comment:
What happens if `uuid_v7` is enabled but `uuid_unstable` is not set? As it stands, it fails to compile with a confusing error message.
Let's handle that case explicitly and emit a compiler error that says "you must pass `uuid_unstable` if `uuid_v7` has been selected. See https://github.com/LukeMathWalker/tracing-actix-web/blob/1a856f8318846c8df4100540051f210ec756fd14/src/lib.rs#L369 as a reference.

Reply:
Hmm, I was planning to do the same, but I instead documented it everywhere. I'll add it